### PR TITLE
Drop duplicate includedir from krb5.conf

### PR DESCRIPTION
**Pull request by tjaalton**
SSSD already provides a config snippet which includes SSSD_PUBCONF_KRB5_INCLUDE_D_DIR, and having both breaks Java.

https://pagure.io/freeipa/issue/9267

### CODE REVIEW

**Comment by abbra** [comment]
I launched full test suite to see how things stand with the removal of `includedir` and relaunched few Azure Ci tests that failed.

File: freeipa.spec.in [hunk 1]
```diff
@@ -1251,10 +1251,8 @@ if [ $1 -gt 1 ] ; then
     test -f '/var/lib/ipa-client/sysrestore/sysrestore.index' && restore=$(wc -l '/var/lib/ipa-client/sysrestore/sysrestore.index' | awk '{print $1}')
 
     if [ -f '/etc/sssd/sssd.conf' -a $restore -ge 2 ]; then
-        if ! grep -E -q '/var/lib/sss/pubconf/krb5.include.d/' /etc/krb5.conf  2>/dev/null ; then
-            echo "includedir /var/lib/sss/pubconf/krb5.include.d/" > /etc/krb5.conf.ipanew
-            cat /etc/krb5.conf >> /etc/krb5.conf.ipanew
-            mv -Z /etc/krb5.conf.ipanew /etc/krb5.conf
+        if grep -E -q '/var/lib/sss/pubconf/krb5.include.d/' /etc/krb5.conf  2>/dev/null ; then
+            sed -i '/includedir \/var\/lib\/sss\/pubconf\/krb5.include.d/d' /etc/krb5.conf
```
Comment (rcritten) [inline freeipa.spec.in:1255]: Only a suggestion, but if you change the delimiter then you don't have to escape the directory and IMHO it's easier to read. Here I'm using semi-colon instead:

`sed -i ';includedir /var/lib/sss/pubconf/krb5.include.d;d' /etc/krb5.conf`

**Comment by rcritten** [comment]
Other than one optional suggestion, this change LGTM.

I tested upgrades and new installs.

File: freeipa.spec.in (hunk 1 above)
Comment (flo-renaud) [reply freeipa.spec.in]: I believe the right syntax to define an alternate separator is `sed -i '\;includedir /var/lib/sss/pubconf/krb5.include.d;d' /etc/krb5.conf`

Comment (tjaalton) [reply freeipa.spec.in]: oh indeed, tested locally now

**Comment by flo-renaud** [comment]
I'm afraid the fix needs to be more complex. In my fedora 37 machine, if I install the freeipa-client package it doesn't install sssd-krb5 by default. This means that `/etc/krb5.conf.d/enable_sssd_conf_dir` does not exist and there is no snippet including `/var/lib/sss/pubconf/krb5.include.d/`.

**Comment by abbra** [comment]
Since IPA client should require krb5 functionality from SSSD (we generate keytabs!) we have to depend on it.
I am a bit surprised that `sssd-krb5` is not required by the `sssd-ipa` but for us it is easier to fix it with `freeipa-client` dependencies rather than expecting `sssd` to fix their setup.

**Comment by tjaalton** [comment]
> 

done

**Comment by flo-renaud** [comment]
Thanks for the update, LGTM.

Test upgrade of the client packages:
- [x] install freeipa-client 4.10.1-1. `/etc/krb5.conf` contains a single includir, `includedir /etc/krb5.conf.d/`
- [x] update freeipa-client package to the version with the patch. `sssd-krb5` package is installed as a dependency. `/etc/krb5.conf` contains a single includedir, `includedir /etc/krb5.conf.d/`, which contains the file `enable_sssd_conf_dir` with `includedir /var/lib/sss/pubconf/krb5.include.d/`.
- [x] install the client with `ipa-client-install`. `/etc/krb5.conf` still contains a single includir, `includedir /etc/krb5.conf.d/`
- [x] uninstall the client. 

Test upgrade of an existing installation:
- [x] install freeipa-client 4.10.1-1. `/etc/krb5.conf` contains a single includir, `includedir /etc/krb5.conf.d/`
- [x] install the client with ipa-client-install. `/etc/krb5.conf.d` contains 2 includedir (`/var/lib/sss/pubconf/krb5.include.d/` has been added by the installer).
- [x] update freeipa-client package to the version with the patch. sssd-krb5 is installed as a dependency. `/etc/krb5.conf` contains a single includedir, `/etc/krb5.conf.d/`, and `/etc/krb5.conf.d/enable_sssd_conf_dir` contains `includedir /var/lib/sss/pubconf/krb5.include.d/`.
- [x] uninstall the client.

Test upgrade of existing installation with double includedir:
- [x] install freeipa-client 4.10.1-1 and sssd-krb5. `/etc/krb5.conf` contains a single includir, `includedir /etc/krb5.conf.d/` and `/etc/krb5.conf.d/enable_sssd_conf_dir` contains `includedir /var/lib/sss/pubconf/krb5.include.d/`
- [x] install the client with ipa-client-install. `/etc/krb5.conf.d` contains 2 includedir, and `/var/lib/sss/pubconf/krb5.include.d/` is included twice.
- [x] update freeipa-client to the version with the patch. `/etc/krb5.conf` contains a single includedir, and `/etc/krb5.conf.d/enable_sssd_conf_dir` contains `includedir /var/lib/sss/pubconf/krb5.include.d/`.
- [x] uninstall the client

**Comment by flo-renaud** [comment]
master:

* a83ae63578124a6cf101d3609213bedb9a66813d Drop duplicate includedir from krb5.conf

**Pull request closed without merging**